packages/api/src/command/medical/patient/add-coordinates.ts

```diff
@@ -0,0 +1,86 @@
+import { Address, combineAddresses } from "../../../domain/medical/address";
+import { Patient } from "../../../domain/medical/patient";
+import { Product } from "../../../domain/product";
+import { AddressGeocodingResult, buildAddressText } from "../../../external/aws/address";
+import { EventTypes, analytics } from "../../../shared/analytics";
+import { capture } from "@metriport/core/util/notifications";
+import { addGeographicCoordinates } from "../address/generate-geocoded-addresses";
+
+const ADDRESS_MATCH_RELEVANCE_THRESHOLD = 0.9;
+
+type AddressBelowThreshold = {
+  relevance: number;
+  address: Address;
+  suggested: string;
+};
+
+/**
+ * Updates the addresses with geographic coordinates and optionally reports low relevance score addresses to the cx.
+ *
+ * @param addresses - a list of Address objects.
+ * @param reportRelevance - optional, boolean to indicate whether to report a bad address to the cx. Defaults to false.
+ * @returns - a list of Address objects with updated coordinates.
+ */
+export async function addCoordinatesToAddresses({
+  addresses,
+  patient,
+  reportRelevance = false,
+}: {
+  addresses: Address[];
+  patient: Partial<Patient>;
+  reportRelevance?: boolean;
+}): Promise<Address[]> {
+  const geocodedAddressResults = await addGeographicCoordinates(addresses);
+  if (reportRelevance) reportLowRelevance(geocodedAddressResults, patient);
+
+  const updatedAddresses = geocodedAddressResults.map(p => p.address);
+  const addressesWithCoordinates = combineAddresses(updatedAddresses, addresses);
+  return addressesWithCoordinates;
+}
+
+export function reportLowRelevance(
+  addresses: AddressGeocodingResult[],
```

Comment:
nit: could just have this in `addGeographicCoordinates` following each call to `geocodeAddress`
Seems like an unnessecary loop again through the addresses and I think it would be simpler. Plus you could change the type of `addGeographicCoordinates` to return just an address and then also get rid of the other looping op here:

`  const updatedAddresses = geocodedAddressResults.map(p => p.address);`

Reply:
nice!